### DISCUSSION TITLE
Is it possible to limit the number of detectable objects in the detection phase?

### DISCUSSION THREAD

**Discussion by AlexPasqua** [PaddlePaddle/PaddleOCR#13272] [Q&A]
I'd like to detect only a limited number of objects (1, in particular), as sometimes my model outputs multiple boxes, but the text recognizer only finds text in one, so it assigns empty string to the others. The confidence for those empty strings is NaN (bug?), so I'm manually filtering them out.

Is there a way to tell the text detector to only output one box?
Similarly to YOLO, where it's possible to limit the number of detectable objects.

I look forward for your insights 😄

**Comment by UserWangZz** [comment]
You can increase the threshold when filtering detection frames to reduce the number of detection frames.

**Reply by AlexPasqua** [reply]
> You can increase the threshold when filtering detection frames to reduce the number of detection frames.

Do you mean manually, as I'm doing now?
Or is there a way to do it directly in paddle? (e.g., passing a threshold value to the detection function).

If you mean the latter, could you provide a minimum code example, please? :)

**Reply by UserWangZz** [reply]
You can find the args named `det_db_box_thresh` in this doc when you use tools/infer/predic_det.py or predict_sys.py.
https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/inference_args_en.md


And this args also in db_postprocess.py, named box_thresh. this mean you can set it in the postprocess with config files.
https://github.com/PaddlePaddle/PaddleOCR/blob/446f1cffbde51766bcc9a0eda6bec9128df5e438/ppocr/postprocess/db_postprocess.py#L29-L57

**Reply by AlexPasqua** [reply]
> You can find the args named `det_db_box_thresh` in this doc when you use tools/infer/predic_det.py or predict_sys.py. https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/inference_args_en.md
> 
> And this args also in db_postprocess.py, named box_thresh. this mean you can set it in the postprocess with config files.
> 
> https://github.com/PaddlePaddle/PaddleOCR/blob/446f1cffbde51766bcc9a0eda6bec9128df5e438/ppocr/postprocess/db_postprocess.py#L29-L57

But this does not guarantee that you'll get only one box, right? You might still get multiple ones, or am I mistaken?
If I'm mistaken, what values should I set for `det_db_box_thresh` or `box_thresh` to get maximum 1 box? They are float values.

Also, can the parameter `max_candidates` be used for this purpose?

**Reply by UserWangZz** [reply]
A reasonable value of det-d x_box_thresh can help post-processing filter out detection boxes that do not meet the requirements. The model outputs multiple detection boxes for the same text instance because some overlapping boxes have high confidence, which makes it impossible for post-processing to discard them. You can try to increase the value of det-d x_box_thresh.
Max_candidates, if I understand correctly, the meaning of this parameter should be the maximum number of candidate boxes selected during NMS execution

**Comment by AlexPasqua** [comment]
TL;DR it's not possible to explicitly limit the number of boxes to a specific number, but only play with the thresholds to reduce the number of boxes (without actual guarantees on the number itself).
(@UserWangZz correct me if I'm wrong)

I'm closing this discussion.
